students/CarolF/module05/MailRoom_3_CFarris.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env Python3
+
+################################
+##Created By: Carol Farris
+##Date: February 13th 2018
+##Goal: MailRoom Part 3 Assignment. 
+##Changes: Moved around code from Mailroom2 version to improve testability (for Mailroom 4).
+##Added try catch exception to while loop managing selection at bottom. Couldn't find a good   
+##place to add comprehension--Will try in Mailroom 4 after I have refactored code a bit more. 
+################################
+
+
+usersChoice = {1: 'Send Thank You', 2: 'Create Report', 3 : 'Send letters to everyone', 4: 'Quit'}
+donorDict = {'Peter_Piper' : [100.00, 200.00 , 3000.00], 'Lucille_Ball': [100.00,200.00], 'Barak_Obama': [100.00,2000.00,3000.00],
+             'Mary_Poppins': [1000, 2000.5], 'Bill_Gates': [100.0, 2000.0, 1000.0]} 
+#For Mailroom 4, put donorDict into function to create dummy donor list. then pass to needed functions 
+             
+
+
+def findDonorName():
+    """This function identifies the donor information to be used"""
+    donorName = str(input("Please provide the donor name for the thank you or type 'list' to see the list of donors >"))
+
+    while donorName == 'list': #as long as user types list, it will return the donor list
+        for key in donorDict:  print(key) #dictionary version to provide list
+        donorName = str(input("Please enter the donor name to compose the thank you >"))
+
```

Comment:
List comprehensions can pretty much be put wherever you have a for loop: 

    [key for key in donorDict.keys()]